core/services/offchainreporting2relay/delegate.go

```diff
@@ -0,0 +1,275 @@
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/core/services/offchainreporting2"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/services/relay"
+	"github.com/smartcontractkit/chainlink/core/services/relay/solana"
+	"github.com/smartcontractkit/chainlink/core/services/telemetry"
+	ocr "github.com/smartcontractkit/libocr/offchainreporting2"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+type Delegate struct {
+	db                    *sqlx.DB
+	jobORM                job.ORM
+	keyStore              keystore.OCR2
+	pipelineRunner        pipeline.Runner
+	peerWrapper           *ocrcommon.SingletonPeerWrapper
+	monitoringEndpointGen telemetry.MonitoringEndpointGenerator
+	chainSet              evm.ChainSet
+	lggr                  logger.Logger
+	relays                relay.Relays
+}
+
+var _ job.Delegate = (*Delegate)(nil)
+
+func NewDelegate(
+	db *sqlx.DB,
+	jobORM job.ORM,
+	keyStore keystore.OCR2,
+	pipelineRunner pipeline.Runner,
+	peerWrapper *ocrcommon.SingletonPeerWrapper,
+	monitoringEndpointGen telemetry.MonitoringEndpointGenerator,
+	chainSet evm.ChainSet,
+	lggr logger.Logger,
+	relays relay.Relays,
+) *Delegate {
+	return &Delegate{
+		db,
+		jobORM,
+		keyStore,
+		pipelineRunner,
+		peerWrapper,
+		monitoringEndpointGen,
+		chainSet,
+		lggr,
+		relays,
+	}
+}
+
+func (d Delegate) JobType() job.Type {
+	return job.OffchainReporting2Relay
+}
+
+func (Delegate) OnJobCreated(spec job.Job) {}
+func (Delegate) OnJobDeleted(spec job.Job) {}
+
+func (Delegate) AfterJobCreated(spec job.Job)  {}
+func (Delegate) BeforeJobDeleted(spec job.Job) {}
+
+func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.Service, err error) {
+	if jobSpec.Offchainreporting2OracleSpec == nil {
+		return nil, errors.Errorf("offchainreporting.Delegate expects an *job.Offchainreporting2OracleSpec to be present, got %v", jobSpec)
+	}
+	spec := jobSpec.Offchainreporting2OracleSpec
+
+	chain, err := d.chainSet.Get(jobSpec.Offchainreporting2OracleSpec.EVMChainID.ToInt())
+	if err != nil {
+		return nil, err
+	}
+
+	ocrdb := offchainreporting2.NewDB(d.db.DB, spec.ID)
+
+	// TODO [relay]: make a relay choice depending on job spec
+	relay := d.relays.Solana
+	relayOCR2Service := relay.NewOCR2Service(solana.OCR2ServiceConfig{
+		NodeURL: "", // TODO [relay]: add validator url
+		Address: spec.ContractAddress.String(),
+		JobID:   spec.ID,
+	})
+	services = append(services, relayOCR2Service)
+
+	var peerID p2pkey.PeerID
+	if spec.P2PPeerID != nil {
+		peerID = *spec.P2PPeerID
+	} else {
+		peerID = chain.Config().P2PPeerID()
+	}
+	peerWrapper := d.peerWrapper
```

Comment:
just a heads up I'm going to port a bunch of the helpers/cleanup I did in ccip for this scaffolding code into core to simplify/reuse all this stuff. I can do it as part of the relay job impl though

Reply:
Yeah, I didn't plan to completely implement the new job type in this PR but just focused on the interface and job delegate.